base_ui.py

```diff
@@ -0,0 +1,278 @@
+from abc import abstractmethod, ABCMeta
```

Comment:
This entire thing looks quite nice, but I thought it could also serve the purpose of not having to define so many "empty" `def func: pass` methods on a CLI manager subclass, relying on the base class having those already, and then only defining those methods that actually do something. However, not marking the methods as abstract could lead to a potential new future subclass forgetting to implement a handler. I'm not entirely sure which solution is best here, this "politically correct" one, where all abstract methods have to be stubbed with empty methods either way, or the "clean code" one where the base class only marks essential methods as abstract, and then the CLI manager only defines those methods that are needed, relying on the base class having the rest of the empty methods already defined. What are your thoughts on this?

Reply:
I see, that's fair enough. Thank you :slightly_smiling_face: